### PR TITLE
Updated to work with alpha7

### PR DESCRIPTION
**Pull request by jasonwynn10**
Made PhpStorm happy with packets
Added lots of PhpDoc Comments

If there's anything else needed to be done before a merge is allowed, please tell me so I can make the changes.

### CODE REVIEW

**Comment by Falkirks** [comment]
Doesn't look to me like any changes will be needed. I will look more later.

File: src/specter/network/SpecterInterface.php [hunk 1]
```diff
@@ -116,17 +121,17 @@ public function putPacket(Player $player, DataPacket $packet, $needACK = false,
                     }
                     break;
                 case BatchPacket::class:
+                	/** @var BatchPacket $packet */
                     $packet->offset = 1;
                     $packet->decode();
 
                     $str = zlib_decode($packet->payload, 1024 * 1024 * 64); //Max 64MB
```
Comment (jasonwynn10) [inline src/specter/network/SpecterInterface.php]: this is causing issues for some reason...

Comment (Falkirks) [inline src/specter/network/SpecterInterface.php]: What issues?

Comment (jasonwynn10) [inline src/specter/network/SpecterInterface.php]: I tested the plugin on my server and when I tried to make a specter, it kept giving an error message

Comment (Falkirks) [inline src/specter/network/SpecterInterface.php]: @jasonwynn10 It would be helpful if you could provide the error message.

Comment (jasonwynn10) [inline src/specter/network/SpecterInterface.php]: I'm looking to reproduce it again myself now

**Comment by jasonwynn10** [comment]
These are the errors I get when testing
screenshot 33
screenshot 34
screenshot 35

**Comment by Falkirks** [comment]
Are you running the latest version from GitHub?

**Comment by jasonwynn10** [comment]
I'm running the latest version modified for alpha7

**Comment by jasonwynn10** [comment]
In the packet's payload, I found this data:
```

ÿö
```
This was running on a default server setup, so I don't know what changed, but I know that something did

**Comment by SOF3** [comment]
On pmmp/PocketMine-MP@2103c98:
```
s s alice
object(pocketmine\network\mcpe\protocol\BatchPacket)#9675 (5) {
  ["payload"]=>
  string(5) "....""
  ["compressionLevel"]=>
  int(0)
  ["isEncoded"]=>
  bool(true)
  ["offset"]=>
  int(0)
  ["buffer"]=>
  string(34) "fe7801010500faff040a86a522033f015c"
}
[14:33:49] [Server thread/CRITICAL]: Unhandled exception executing command 's s alice' in specter: zlib_decode(): data error
[14:33:49] [Server thread/CRITICAL]: ErrorException: "zlib_decode(): data error" (EXCEPTION) in "plugins/Specter_src/src/specter/network/SpecterInterface" at line 129
[14:33:49] [Server thread/DEBUG]: #0 (): pocketmine\{closure}(integer 2, string zlib_decode(): data error, string C:\sofe\sand\pm1\plugins\Specter_src\src\specter\network\SpecterInterface.php, integer 129, array Array())
[14:33:49] [Server thread/DEBUG]: #1 plugins/Specter_src/src/specter/network/SpecterInterface(129): zlib_decode(string ....", integer 67108864)
```
I added a var_dump($packet) before decoding the packet.

**Comment by jasonwynn10** [comment]
Thank you @SOF3

**Comment by jasonwynn10** [comment]
Ready to merge @Falkirks

**Pull request merged**